cmd/testcase-extractor/main.go

```diff
@@ -0,0 +1,386 @@
+		go compileTestConfig(dmnFile, searchDir, testConfigsCh)
+	}
+
+	testConfigs := make([]TestConfig, 0)
+	for _ = range len(dmnFiles) {
+		testConfigs = append(testConfigs, <-testConfigsCh)
+	}
+
+	sort.Slice(testConfigs, func(i, j int) bool {
+		return *testConfigs[i].Model.Name < *testConfigs[j].Model.Name
+	})
+
+	return testConfigs
+}
+
+func compileTestConfig(
+	dmnFile string,
+	searchDir string,
+	testConfigsCh chan TestConfig,
+) {
+	testCaseDir := strings.TrimPrefix(filepath.Dir(dmnFile), searchDir)
+	testCaseFile := strings.TrimSuffix(
+		dmnFile,
+		filepath.Ext(dmnFile),
+	) + TestCaseFileSuffix
+
+	tckTestCasesCh := make(chan *TckTestCases)
+	go extractTckTestCases(testCaseFile, tckTestCasesCh)
+
+	tckDecisionsCh := make(chan []*TckDecision)
+	go extractTckDecisions(dmnFile, tckDecisionsCh)
+
+	tckTestCases, tckDecisions := <-tckTestCasesCh, <-tckDecisionsCh
+
+	tckDecisionsMap := mapTckDecisionsByName(tckDecisions)
+
+	testCases := make([]TestCase, 0)
+	for _, tckTestCase := range tckTestCases.TestCases {
+		results := make([]TestCaseResult, 0)
+		for _, tckResult := range tckTestCase.Results {
+			tckDecision := tckDecisionsMap[*tckResult.Name]
+			result := TestCaseResult{
+				FeelExpression: tckDecision.LiteralExpression,
+				Expected:       tckResult.Expected.Value,
+			}
+			results = append(results, result)
+		}
+
+		testCase := TestCase{
+			Id:          tckTestCase.Id,
+			Description: tckTestCase.Description,
+			Results:     results,
+		}
+		testCases = append(testCases, testCase)
+	}
+
+	testConfigsCh <- TestConfig{
+		Model: TestConfigModel{
+			tckTestCases.ModelName,
+			&testCaseDir,
+		},
+		TestCases: testCases,
+	}
+}
+
+func mapTckDecisionsByName(tckDecisions []*TckDecision) map[string]*TckDecision {
+	resultsMap := make(map[string]*TckDecision)
+	for _, tckDecision := range tckDecisions {
+		resultsMap[*tckDecision.Name] = tckDecision
+	}
+
+	return resultsMap
+}
+
+func saveTestConfigs(testConfigs []TestConfig, outputFile *os.File) {
+	yamlData, err := yaml.Marshal(&testConfigs)
+	if err != nil {
+		printError("Error marshaling to YAML: %v\n", err)
+		os.Exit(1)
+	}
+
+	_, err = outputFile.Write(yamlData)
+	if err != nil {
+		printError("Error writing YAML file: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func printError(format string, a ...any) {
+	_, _ = fmt.Fprintf(os.Stderr, format, a...)
+}
+
+func isOutputPiped() bool {
+	fileInfo, err := os.Stdout.Stat()
+	if err != nil {
+		printError("Error checking pipe: %v\n", err)
+	}
+	// Check if the file mode indicates a named pipe (FIFO)
+	return (fileInfo.Mode() & os.ModeNamedPipe) != 0
+}
+
+func printExtractionStats(
+	outputFile *os.File,
+	countOfFeelExpressions int,
+	countOfFiles int,
+	duration time.Duration,
+) {
+
+	outputFileInfo := ""
+	if outputFile != os.Stdout {
+		outputFilenameAbs, _ := filepath.Abs(outputFile.Name())
+		outputFileInfo = fmt.Sprintf(
+			" to file "+
+				"\n\t%v",
+			outputFilenameAbs,
+		)
+	}
+
+	fmt.Printf(
+		"\n"+
```

Comment:
fyi for more complicated templating go has [text/template](https://pkg.go.dev/text/template) (you dont have to change this now)